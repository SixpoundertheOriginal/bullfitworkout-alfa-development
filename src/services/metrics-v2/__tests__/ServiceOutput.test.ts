import { describe, it, expect } from 'vitest';
import { getMetricsV2, InMemoryRepoStub } from '../index';
import { withoutVolatile } from '../../../../tests/helpers/stableSnapshot';

describe('ServiceOutput shape (v2)', () => {
  it('returns a versioned, canonical structure', async () => {
    const out = await getMetricsV2(InMemoryRepoStub, 'u1', { from: new Date(), to: new Date() });
    expect(out.meta.version).toBe('v2');
    expect(out.series.base.length).toBe(out.series.base.length); // sanity
    expect(withoutVolatile(out)).toMatchSnapshot();
    });
});
