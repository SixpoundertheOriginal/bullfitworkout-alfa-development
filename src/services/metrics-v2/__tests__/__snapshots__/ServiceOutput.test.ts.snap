// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ServiceOutput shape (v2) > returns a versioned, canonical structure 1`] = `
{
  "meta": {
    "inputs": {
      "tz": "Europe/Warsaw",
      "units": "kg|min",
    },
    "version": "v2",
  },
  "metricKeys": [
    "tonnage_kg",
    "sets_count",
    "reps_total",
    "density_kg_per_min",
    "cvr",
    "workouts",
    "duration_min",
    "avgRestSec",
    "setEfficiencyKgPerMin",
  ],
  "perWorkout": [],
  "prs": [],
  "series": {
    "avgRestSec": [],
    "cvr": [],
    "density_kg_per_min": [],
    "duration_min": [],
    "reps_total": [],
    "setEfficiencyKgPerMin": [],
    "sets_count": [],
    "tonnage_kg": [],
    "workouts": [],
  },
  "timePeriodAverages": undefined,
  "timingMetadata": {
    "coveragePct": 0,
    "quality": "low",
  },
  "totals": {
    "avgRestSec": 0,
    "density_kg_per_min": 0,
    "duration_min": 0,
    "reps_total": 0,
    "setEfficiencyKgPerMin": 0,
    "sets_count": 0,
    "timingCoveragePct": 0,
    "tonnage_kg": 0,
    "workouts": 0,
  },
}
`;
